workshops/2021-16-02_intro_to_github/TOG_git_tutorial.Rmd

```diff
@@ -0,0 +1,396 @@
+
+Even for private solo projects, it’s a good idea to push your work to a remote location for peace of mind. Why? Because it’s fairly easy to screw up your local Git repository, especially when you’re new at this. The good news is that often only the Git infrastructure is borked up. Your files are just fine! Which makes your Git pickle all the more frustrating. There are official Git solutions to these problems, but they might require expertise and patience you can’t access at 3a.m. If you’ve recently pushed your work to GitHub, it’s easy to grab a fresh copy, patch things up with the changes that only exist locally, and get on with your life.
+
+We target GitHub – not Bitbucket or GitLab – for the sake of specificity. However, all the big-picture principles and even some mechanics will carry over to these alternative hosting platforms.
+
+
+### Wait, so what is what, exactly? 
+
+**_Version control_** is the general term used to refer to software systems that helps you "manage" your files, most commonly being code bases. As already mentioned, Git is just one of many version control systems out there. Other examples include [Mercurial](https://www.mercurial-scm.org/), [Perforce](https://www.perforce.com/), [Subversion](https://subversion.apache.org/), etc.
+
+**_Git_** is THE version control software. Both the GitHub server and your computer run it. It is what enables your laptop to talk with the remote repository stored on GitHub, and enables version control features like pushing and pulling code changes, keeping history, etc. 
+
+**_GitHub_** is the service in the cloud that stores your remote repository. When you're working in a group, you and your collaborators will independently interact with GitHub in order to pull/publish your latest code changes. It will hold the most up-to-date "truth" of your project directory with each of you working away on your local machines. 
+
+**_Git Client_** is a piece of software that provides a graphical user interface that facilitates your interaction with Git. In our case, **we will be using RStudio's built-in Git Client**. Other options include [SourceTree](https://www.sourcetreeapp.com/) and [GitHub Desktop](https://desktop.github.com/). And of course, the terminal is the most powerful (also the most difficult) way to use Git. 
+
+
+## Part 2: Set up your environment
+
+Before we can start using Git, your computer must be equipped with the appropriate softwares. In this section, we will walk you through: 1. Installing Git, 2. Registering for a GitHub account, 3. Setting up an SSH key, 4. Talking to Git via RStudio. 
+
+### 1. Install Git (contains excerpts from [Ch. 7 of Happy Git and GitHub for the useR](http://happygitwithr.com/install-git.html))
```

Comment:
Potentially link all standalone installation/download steps at the beginning under a "requirements" section